pkg/controller/multi_writer_at.go

```diff
@@ -45,18 +47,26 @@ func (m *MultiWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
 			if err != nil {
 				errored = true
 				errs[index] = err
+			} else {
+				succeeded = true
 			}
 			wg.Done()
 		}(i, w)
 	}
 
 	wg.Wait()
+
+	// Return length of p if at least one WriteAt succeeded
+	if succeeded {
+		n = len(p)
+	}
+
```

Comment:
```go
n: = max(written)
var err *MultiWriterError
total := sum(written)
if total < (len(p) * len(wri) {
    err = &MultiWriterError{ Writers: m.writers, Errors:  errs }
}
return n, err
```

This makes the code a little nicer to read, you can either use the total approach or check the errors array explicitly for non null.

Reply:
I had a discussion with Joshua and we came up with a solution that doesn't have unsafe variable access from goroutines and doesn't compute the max and sum.